### ISSUE TITLE
Allow relaxed shorthand syntax for "margin"

### ISSUE THREAD

**Issue by masi** [brigade/scss-lint#392]
IMHO "0 1px 0" is less readable than "0 1px 0 1px". So I'd appreciate an option that allows these variations:

margin: 0
margin: 0 1px
margin: 0 1px 2px 3px

Regards,
Masi

**Comment by chris-dura** [comment]
I do believe that is exactly what https://github.com/causes/scss-lint/blob/master/lib/scss_lint/linter/README.md#shorthand does?

With that linter enabled both `0 1px 0` and `0 1px 0 1px` should throw errors, recommending the more concise `0 1px`.

**Comment by masi** [comment]
No, my issue is with the current behaviour of Shorthand.

CSS allows for the follwoing two readable shorthands:

margin: top/right/bottom/left
margin: top/bottom right/left

But it allows also for that one (using 3 values):

margin: top right/left bottom

I don't like the last one. So I would like to configure SCSS-Lint to allow the first two ones but report an error for the last one.

**Comment by sds** [comment]
I'd be open to a pull request adding an `allowed_shorthands` property to the `Shorthand` linter which allows you to specify which shorthands are allowed, e.g.

```yaml
linters:
  Shorthand:
    allowed_shorthands: [1, 4]
```

...would allow for shorthands where you either specify just one number for all sides or all numbers explicitly.
